visualization/app/codeCharta/util/customConfigHelper.ts

```diff
@@ -43,65 +51,146 @@ export class CustomConfigHelper {
+			if (
+				customConfig.name === configName &&
+				customConfig.mapSelectionMode === mapSelectionMode &&
+				customConfig.assignedMaps.join("") === selectedMaps.join("")
+			) {
+				return true
+			}
+		}
+
+		return false
+	}
+
+	static getCustomConfigs(): Map<string, CustomConfig> {
+		return this.customConfigs
 	}
 
-	static hasCustomConfig(mapSelectionMode: CustomConfigMapSelectionMode, selectedMaps: string[], viewName: string): boolean {
-		const customConfigIdentifier = createCustomConfigIdentifier(mapSelectionMode, selectedMaps, viewName)
+	static importCustomConfigs(content: string) {
+		const importedCustomConfigsFile: CustomConfigsDownloadFile = JSON.parse(content, stateObjectReviver)
+
+		for (const exportedConfig of importedCustomConfigsFile.customConfigs.values()) {
+			const alreadyExistingConfig = CustomConfigHelper.getCustomConfigSettings(exportedConfig.id)
 
-		return this.customConfigs.has(customConfigIdentifier)
+			// Check for a duplicate Config by matching checksums
+			if (alreadyExistingConfig) {
+				continue
+			}
+
+			// Prevent different Configs with the same name
+			if (
+				CustomConfigHelper.hasCustomConfigByName(exportedConfig.mapSelectionMode, exportedConfig.assignedMaps, exportedConfig.name)
+			) {
+				exportedConfig.name += ` (${FileNameHelper.getFormattedTimestamp(new Date(exportedConfig.creationTime))})`
+			}
+
+			const importedCustomConfig: CustomConfig = {
+				id: exportedConfig.id,
+				name: exportedConfig.name,
+				creationTime: exportedConfig.creationTime,
+				assignedMaps: exportedConfig.assignedMaps,
+				customConfigVersion: exportedConfig.customConfigVersion,
+				mapChecksum: exportedConfig.mapChecksum,
+				mapSelectionMode: exportedConfig.mapSelectionMode,
+				stateSettings: exportedConfig.stateSettings
+			}
+
+			CustomConfigHelper.addCustomConfig(importedCustomConfig)
+		}
+	}
+
+	static downloadCustomConfigs(
+		customConfigs: Map<string, ExportCustomConfig>,
+		customConfigFileStateConnector: CustomConfigFileStateConnector
+	) {
+		const customConfigsDownloadFile: CustomConfigsDownloadFile = {
+			downloadApiVersion: CUSTOM_CONFIGS_DOWNLOAD_FILE_VERSION,
+			timestamp: Date.now(),
+			customConfigs
+		}
+
+		let fileName = FileNameHelper.getNewTimestamp() + CUSTOM_CONFIG_FILE_EXTENSION
+
+		if (
+			!customConfigFileStateConnector.isDeltaMode() &&
+			customConfigFileStateConnector.getAmountOfUploadedFiles() === 1 &&
+			customConfigFileStateConnector.isEachFileSelected()
+		) {
+			// If only one map is uploaded/present in SINGLE mode, prefix the .cc.config.json file with its name.
+			fileName = `${FileNameHelper.withoutCCJsonExtension(customConfigFileStateConnector.getJointMapName())}_${fileName}`
+		}
+
+		FileDownloader.downloadData(JSON.stringify(customConfigsDownloadFile, stateObjectReplacer), fileName)
+	}
+
+	static createExportCustomConfigFromConfig(customConfig: CustomConfig): ExportCustomConfig {
+		const exportCustomConfig: ExportCustomConfig = {
```

Comment:
That is simply a complete shadow clone like `return { ...customConfig }`, isn't it?